ui/pages/permissions-connect/snaps/snaps-connect/snaps-connect.js

```diff
@@ -0,0 +1,298 @@
+          justifyContent={JustifyContent.center}
+          alignItems={AlignItems.center}
+          paddingLeft={4}
+          paddingRight={4}
+          paddingTop={8}
+          width={BlockSize.Full}
+          style={{ overflowY: 'hidden' }}
+        >
+          <Text paddingBottom={2} variant={TextVariant.headingLg}>
+            {t('connectionRequest')}
+          </Text>
+          <Text variant={TextVariant.bodyMd} textAlign={TextAlign.Center}>
+            {t('multipleSnapConnectionWarning', [
+              <Text
+                as={ValidTag.Span}
+                key="1"
+                variant={TextVariant.bodyMd}
+                fontWeight={FontWeight.Medium}
+              >
+                {trimmedOrigin}
+              </Text>,
+              <Text
+                as={ValidTag.Span}
+                key="2"
+                variant={TextVariant.bodyMd}
+                fontWeight={FontWeight.Medium}
+              >
+                {snaps?.length}
+              </Text>,
+            ])}
+          </Text>
+          <Box
+            className="snaps-connect__content__snaps-list"
+            flexDirection={FlexDirection.Column}
+            display={Display.Flex}
+            marginTop={4}
+            width={BlockSize.Full}
+            style={{ overflowY: 'auto', flex: 1 }}
+          >
+            {snaps.map((snap) => (
+              // TODO(hbmalik88): add in the iconUrl prop when we have access to a snap's icons pre-installation
+              <SnapConnectCell
+                key={`snaps-connect-${snap}`}
+                snapId={snap}
+                origin={trimmedOrigin}
+              />
+            ))}
+          </Box>
+        </Box>
+      );
+    } else if (snaps?.length === 1) {
+      return (
+        <Box
+          className="snaps-connect__content"
+          flexDirection={FlexDirection.Column}
+          justifyContent={JustifyContent.center}
+          alignItems={AlignItems.center}
+          height={BlockSize.Full}
+          paddingLeft={4}
+          paddingRight={4}
+        >
+          <Box paddingBottom={2}>
+            <ConnectIcon />
+          </Box>
+          <Text paddingBottom={2} variant={TextVariant.headingLg}>
+            {t('connectionRequest')}
+          </Text>
+          <Text
+            variant={TextVariant.bodyMd}
+            textAlign={TextAlign.Center}
+            padding={[0, 4]}
+          >
+            {t('snapConnectionWarning', [
+              <Text
+                as={ValidTag.Span}
+                key="1"
+                variant={TextVariant.bodyMd}
+                fontWeight={FontWeight.Medium}
+              >
+                {trimmedOrigin}
+              </Text>,
+              <Text
+                as={ValidTag.Span}
+                key="2"
+                variant={TextVariant.bodyMd}
+                fontWeight={FontWeight.Medium}
+              >
+                {getSnapName(snaps?.[0])}
```

Comment:
Might be good to pass `subjectMetadata` to `getSnapName`, if the snap is already installed it will be able to get the name of the snap

Reply:
Do you mean the `targetSubjectMetadata` that the component already receives? If so, that is site metadata, not snap metadata.

Or do you mean the following?

```typescript
const snapId = snaps?.[0];
const snapMetadata = useOriginMetadata(snapId);
const snapName = getSnapName(snapId, snapMetadata);
```

Reply:
```
const subjectMetadata = useSelector((state) => getTargetSubjectMetadata(state, snapId));
const snapName = getSnapName(snapId, subjectMetadata);
```